src/lib/compute.ts

```diff
@@ -0,0 +1,98 @@
+import {Node, Params} from '../types/compute';
+import {PluginsStorage} from '../types/initialize';
+import {PluginParams} from '../types/interface';
+import {ManifestCommon} from '../types/manifest';
+
+/**
+ * Traverses all child nodes based on children grouping.
+ */
+const traverse = async (children: any, params: Params) => {
+  for (const child in children) {
+    await computeNode(children[child], params);
+  }
+};
+
+/**
+ * Appends `default` values to `inputs`.
+ */
+const mergePluginParams = (
+  inputs: PluginParams[],
+  defaults: PluginParams[] | undefined
+) =>
+  inputs.map(input => ({
```

Comment:
Please check the inputs before mapping (e.g., inputs && ) as it fails when running the `mock-observation` plugin, which does not have any input